super_calendar_analytic/models/super_calendar.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Odoo, Open Source Management Solution
+#    This module copyright (C) 2015 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class SuperCalendarConfiguratorLine(models.Model):
+    _inherit = 'super.calendar.configurator.line'
+
+    analytic_field_id = fields.Many2one(
+        comodel_name='ir.model.fields',
+        string='Analytic Account field',
+        domain="['&', ('ttype', '=', 'many2one'), ('model_id', '=', name)]",
+    )
+
+
+class SuperCalendar(models.Model):
+    _inherit = 'super.calendar'
+
+    analytic_id = fields.Many2one(
+        comodel_name='account.analytic.account',
+        string='Analytic Account',
+    )
+
+
+class SuperCalendarConfigurator(models.Model):
+    _inherit = 'super.calendar.configurator'
+
+    @api.multi
+    def _get_record_values_from_line(self, configurator, line):
+        res = super(
+            SuperCalendarConfigurator, self)._get_record_values_from_line(
+            configurator, line
+        )
+        for record in res:
+            f_analytic = (line.analytic_field_id and
```

Comment:
`and` is not required as getting a field from an empty record set should return False